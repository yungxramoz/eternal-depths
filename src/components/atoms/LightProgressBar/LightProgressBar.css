.light-progress-bar {
  display: block;
  position: relative;
  width: 100%;
  height: 18px;
}

.light-progress-track {
  left: 1em;
  right: 1em;
  background-color: dimgrey;
  box-shadow: -2px 2px 1px black;
  background-image: none;
  height: 18px;
}

.light-progress-track > .rpgui-progress-fill {
  top: 0;
  bottom: 0;
  height: 18px;
}

.progress-animation {
  transition: width 0.2s linear;
}
