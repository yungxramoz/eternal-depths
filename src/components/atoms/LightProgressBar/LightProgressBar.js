import React from 'react'
import './LightProgressBar.css'

const LightProgressBar = ({ current, max, type }) => {
  const progress = (current / max) * 100
  let color = 'red'
  switch (type) {
    case 'hp':
      color = 'red'
      break
    case 'xp':
      color = 'blue'
      break
    default:
      break
  }
  return (
    <div className="light-progress-bar">
      <div className="rpgui-progress-track light-progress-track">
        <div
          className={`rpgui-progress-fill ${color} progress-animation`}
          style={{ left: '0px', width: `${progress}%` }}
        ></div>
      </div>
    </div>
  )
}

export default LightProgressBar
