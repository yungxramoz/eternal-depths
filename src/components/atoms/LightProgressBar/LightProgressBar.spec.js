import React from 'react'
import { render } from '@testing-library/react'
import LightProgressBar from './LightProgressBar'

describe('LightProgressBar', () => {
  it('renders correctly HP bar', () => {
    render(<LightProgressBar current={50} max={100} type="hp" />)
    expect(document.body).toMatchSnapshot('LightProgressBar Hp')
  })

  it('renders correctly XP bar', () => {
    render(<LightProgressBar current={50} max={100} type="xp" />)
    expect(document.body).toMatchSnapshot('LightProgressBar Xp')
  })
})
