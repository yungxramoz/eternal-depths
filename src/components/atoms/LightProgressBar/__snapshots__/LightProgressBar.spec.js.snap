// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LightProgressBar renders correctly HP bar: LightProgressBar Hp 1`] = `
<body>
  <div>
    <div
      class="light-progress-bar"
    >
      <div
        class="rpgui-progress-track light-progress-track"
      >
        <div
          class="rpgui-progress-fill red progress-animation"
          style="left: 0px; width: 50%;"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`LightProgressBar renders correctly XP bar: LightProgressBar Xp 1`] = `
<body>
  <div>
    <div
      class="light-progress-bar"
    >
      <div
        class="rpgui-progress-track light-progress-track"
      >
        <div
          class="rpgui-progress-fill blue progress-animation"
          style="left: 0px; width: 50%;"
        />
      </div>
    </div>
  </div>
</body>
`;
